proto/xmtpv4/message_api/message_api.proto

```diff
@@ -116,12 +111,12 @@ message QueryEnvelopesResponse {
   repeated OriginatorEnvelope envelopes = 1;
 }
 
-message PublishEnvelopeRequest {
-  PayerEnvelope payer_envelope = 1;
+message PublishEnvelopesRequest {
+  repeated PayerEnvelope payer_envelopes = 1;
 }
 
-message PublishEnvelopeResponse {
-  OriginatorEnvelope originator_envelope = 1;
+message PublishEnvelopesResponse {
+  repeated OriginatorEnvelope originator_envelopes = 1;
```

Comment:
We could, but without the bytes the client doesn't have a fool-proof way to verify the signature without issuing another request. I suppose instead of sending up `PayerEnvelope` the client can send up bytes for the server to sign? Non-deterministic serialization bites us again

Reply:
Ahh, you're right. We need to get the signed originator envelope back to verify the timestamp, which the client can't derive on its own since it includes the sequence ID and timestamp.

Reply:
We could do something somewhat complicated to save the response bytes:
1. Client uploads payer envelope as bytes
2. Server constructs a receipt payload that contains sequence ID, timestamp, payer envelope hash. Signs that and returns the result to the client.
3. An originator envelope contains both a receipt payload and the payer envelope bytes.

Another approach would be to have the client specify a flag for whether it wants a receipt or not. Risk there is that clients would never set the flag for large payloads, which are the ones servers would most like to drop as they are resource intensive.

Reply:
I think we leave it and just return the full originator envelope. Will make client's lives easier to get the same format from published and queried envelopes, so you can optimistically update your own state as soon as the message is published.